src/Main.elm

```diff
@@ -25,40 +143,939 @@ init =
+
+                                    Post ->
+                                        let
+                                            addressValue =
+                                                inputFields |> Dict.get "address" |> Maybe.withDefault ""
+
+                                            addressErrorMsg =
+                                                "Details Tab: Address is mandatory field"
+                                        in
+                                        if String.isEmpty addressValue then
+                                            Set.insert addressErrorMsg Set.empty
+
+                                        else
+                                            Set.remove addressErrorMsg Set.empty
+
+                                    Phone ->
+                                        let
+                                            errorMsg =
+                                                "Details Tab: Phone is mandatory field"
+                                        in
+                                        if String.isEmpty phoneValue then
+                                            Set.insert errorMsg Set.empty
+
+                                        else
+                                            Set.remove errorMsg Set.empty
+                        in
+                        [ emailError, phoneError, madatoryFieldsErrors ] |> List.foldl (\set acc -> Set.union set acc) Set.empty
+
+                settingsErrors =
+                    -- We'll validate settings only if not inherited
+                    if isSettingsInherited then
+                        Set.empty
+
+                    else
+                        inputFields
+                            |> Dict.foldl
+                                (\key value acc ->
+                                    if List.member key settingsList then
+                                        if String.toInt value == Nothing && String.length value /= 0 then
+                                            Set.insert "Settings Tab: Field should be number" acc
+
+                                        else
+                                            acc
+
+                                    else
+                                        acc
+                                )
+                                Set.empty
+            in
+            Set.union detailsErrors settingsErrors
+    in
+    updateErrors
+
+
+fetchUserGroup : Cmd Msg
+fetchUserGroup =
+    Http.get
+        { url = serverUrl
+        , expect = Http.expectJson GetUserGroup decodeUserGroup
+        }
+
+
+decodeUserGroup : Decoder UserGroup
+decodeUserGroup =
+    Decode.succeed UserGroup
+        |> required "settings" decodeSettings
+        |> required "contact_details" decodeContactDetails
+        |> required "tags" decodeTags
+
+
+decodeSettings : Decoder Settings
+decodeSettings =
+    Decode.succeed Settings
```

Comment:
Q: This approach suffers from being error prone to parse wrong value, if you e.g. swap lines 499 and 500 below. What would you do to minimize the risk of swapping those values?

Reply:
Yeah, its error prone since its based on a order of fields.  I think `oneOf` would do a trick ?